core/problems/problems_test.go

```diff
@@ -0,0 +1,65 @@
+package problems_test
+
+import (
+	"testing"
+
+	"github.com/otrego/clamshell/core/movetree"
+	"github.com/otrego/clamshell/core/problems"
+	"github.com/otrego/clamshell/core/sgf"
+)
+
+func TestFlatten(t *testing.T) {
+
+	testCases := []struct {
+		desc string
+		tp   string
+		sgf  string
+	}{
+		{
+			desc: "entire game flatten",
+			tp:   "0:140",
+			sgf: `(;FF[4]CA[UTF-8]GM[1]DT[2020-08-05]PB[player1]PW[player2]
+				BR[2k] WR[5d] TM[259200]OT[86400 fischer] RE[W+R] SZ[19]
+				KM[6.5] RU[Japanese] ;B[pd] (;W[dc] (;B[qp] (;W[cq] (;B[np]
```

Comment:
I wouldn't try to simplify this game for a unit-test. As is, it
1. Has too many moves.
2. It has mega-unnecessary branching for some reason.